### PR TITLE
R3F/RHS Compat - Add overheating and update mass

### PR DESCRIPTION
**Pull request by Laid3acK**
**When merged this pull request will:**
- Add R3F and RHS lmgs overheating according to the [overheating/CfgWeapons.hpp](https://github.com/acemod/ACE3/blob/master/addons/overheating/CfgWeapons.hpp) and the [Overheating Framework](https://ace3mod.com/wiki/framework/overheating-framework.html).
- Update all R3F weapons `mass`.
- Add `ACE_isBelt`.

### IMPORTANT

- [ ] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3mod.com/).
- [x] Development Guidelines](https://ace3mod.com/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: optionals/compat_rhs_gref3/CfgWeapons.hpp [hunk 1]
```diff
@@ -87,9 +87,11 @@ class CfgWeapons {
     };
 
     class rhs_weap_mg42_base: Rifle_Base_F {
-        ACE_Overheating_allowSwapBarrel = 1;
         ACE_barrelTwist = 305.0;
         ACE_barrelLength = 530.0;
+        GVAR(allowSwapBarrel) = 1;
```
Comment (Salluci) [inline optionals/compat_rhs_gref3/CfgWeapons.hpp:92]: https://ace3mod.com/wiki/development/coding-guidelines.html#macro-usage

Comment (Laid3acK) [reply optionals/compat_rhs_gref3/CfgWeapons.hpp:92]: https://github.com/acemod/ACE3/blob/master/addons/overheating/CfgWeapons.hpp#L48

Comment (Salluci) [reply optionals/compat_rhs_gref3/CfgWeapons.hpp:92]: Will expand to ace_compat_rhs_gref3_allowSwapBarrel here. Different module. No bueno. Use EGVAR macro.

**Comment by Drofseh** [comment]
Is changing other mod's weapon mass within ace's scope?
Could mess up existing missions.

**Review by PabstMirror — CHANGES_REQUESTED** [review]
change `GVAR(x)` to `EGVAR(overheating,x)`

**Pull request closed without merging**